extensions/community/Sprite3D.json

```diff
@@ -73,41 +121,162 @@
+            "     * Apply deep perspective effect",
+            "     */",
+            "    applyDepthPerspective() {",
+            "        if (!this.mesh) return;",
+            "        ",
+            "        const obj = this.object;",
+            "        const yPos = obj.getY ? obj.getY() : 0;",
+            "        const zPos = obj.getZ ? obj.getZ() : 0;",
+            "        ",
+            "        // Calculate scale factor based on depth (perspective)",
+            "        const perspectiveFactor = 1 - (zPos * 0.0001); // Reduce size of distant objects",
+            "        const scaleFactor = Math.max(0.5, Math.min(1.5, perspectiveFactor));",
+            "        ",
+            "        // Apply additional factor based on Y (apparent height)",
+            "        const heightFactor = 1 + (yPos * 0.00005);",
+            "        ",
+            "        // Update scale while maintaining original proportions",
+            "        const frame = this.object.getAnimator().getCurrentFrame();",
+            "        if (frame) {",
+            "            const image = frame.texture.map.image;",
+            "            const width = image.width;",
+            "            const height = image.height;",
+            "            ",
+            "            this.mesh.scale.set(",
+            "                width * scaleFactor * heightFactor,",
+            "                height * scaleFactor * heightFactor,",
+            "                1",
+            "            );",
+            "        }",
+            "    }",
+            "    ",
             "    updateFrame() {",
             "        const frame = this.object.getAnimator().getCurrentFrame();",
             "        if (!frame) {",
             "            return;",
             "        }",
             "        const material = frame.texture;",
-            "",
+            "        ",
+            "        // Update basic settings",
+            "        material.alphaTest = 0.5;",
+            "        material.depthWrite = true;",
+            "        material.depthTest = true;",
+            "        material.transparent = true;",
+            "        material.side = THREE.DoubleSide;",
+            "        material.shadowSide = THREE.FrontSide;",
+            "        material.toneMapped = true;",
+            "        material.fog = true;",
+            "        material.premultipliedAlpha = true;",
+            "        material.depthFunc = THREE.LessEqualDepth;",
+            "        material.polygonOffset = true;",
+            "        material.polygonOffsetFactor = -1;",
+            "        material.polygonOffsetUnits = -1;",
+            "        ",
+            "        // Professional visual effects",
+            "        if (material.emissive) {",
+            "            material.emissive = new THREE.Color(0x111111);",
+            "            material.emissiveIntensity = 0.2;",
+            "        }",
```

Comment:
This will override the existing material if any.
Not recommended to set these values in stone.